TrafficSimulator/Assets/Prefabs/Car/AutoDrive.cs

```diff
@@ -159,16 +169,55 @@ void Update()
             {
                 P_MoveToNextPosition();
             }
+            updateOccupiedNodes();
+        }
+
+        // Update the list of nodes that the vehicle is currently occupying
+        private void updateOccupiedNodes()
+        {
+            foreach (LaneNode node in _occupiedNodes)
+            {
+                node.UnsetVehicle(_vehicle);
+            }
+            _occupiedNodes = getOccupiedNodes();
+            foreach (LaneNode node in _occupiedNodes)
+            {
+                node.SetVehicle(_vehicle);
+            }
+        }
+
+        // Get the list of nodes that the vehicle is currently occupying by moving backwards from the current position until out of vehicle bounds
+        private List<LaneNode> getOccupiedNodes()
+        {
+            List<LaneNode> nodes = new List<LaneNode>();
+            LaneNode node = _currentNode;
+            while (node != null && Vector3.Distance(node.Position, transform.position) <= _vehicleLength/1.5)
+            {
+                nodes.Add(node);
+                node = node.Prev;
+            }
+            try {
```

Comment:
No need for a try/catch here. Just set the node to `_currentNode.Next`, then have a while loop in the same fashion as the previous one. If `Next` is null then the while loop will never be entered